catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -867,13 +867,31 @@ private void generateMetacardAndContentItems(StorageRequest storageRequest,
                 mimeTypeRaw = guessMimeType(mimeTypeRaw, contentItem.getFilename(), tmpPath);
 
                 String fileName = updateFileExtension(mimeTypeRaw, contentItem.getFilename());
-                Metacard metacard = generateMetacard(mimeTypeRaw,
-                        contentItem.getId(),
-                        fileName,
-                        size,
-                        (Subject) storageRequest.getProperties()
-                                .get(SecurityConstants.SECURITY_SUBJECT),
-                        tmpPath);
+
+                Metacard metacard;
+
+                if (contentItem.getMetacard() != null) {
```

Comment:
This has me a little bit worried, the more I think about it. Currently, we use the generated metacard to determine whether or not the user and system has sufficient privileges to store that content item. With this change, I could now craft a metacard with any sort of security profile, and store any content item, regardless of content. 

We do allow you to craft any metacard and store any metacard in the catalog. I just feel like the difference with that, is you're storing the thing that we're checking. With this, you're storing a metacard along with a content item that could be entirely unrelated.